### PR TITLE
Add types for didyoumean.js

### PR DESCRIPTION
**Pull request by jamesgeorge007**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`

### CODE REVIEW

File: types/didyoumean/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// Type definitions for didyoumean 1.2.1
```
Comment (andrewbranch) [inline types/didyoumean/index.d.ts]: FYI, this is what’s causing the builds to fail—the version number should only be major/minor, no patch number.

**Comment by andrewbranch** [comment]
/azp run DefinitelyTyped.BenchmarkPR

(I'm testing making the perf results a little quieter on updates, ignore me 😄)

**Comment by jamesgeorge007** [comment]
cc @amcasey

File: types/didyoumean/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+export = didYouMean;
+
+declare function didYouMean(str: string, list: string[], key?: string): string | string[];
+
+declare namespace didYouMean {
+    let caseSensitive: boolean;
+
+    let nullResultValue: any;
+
+    let prototype: {
```
Comment (amcasey) [inline types/didyoumean/index.d.ts]: I don't believe you need to explicitly declare `prototype` - you should get it from the merged function declaration.

**Comment by jamesgeorge007** [comment]
@amcasey Hope it's fine now :thinking:

**Review by amcasey — APPROVED** [review]
LGTM.  Thanks!

**Comment by iabdo9** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/fc9a888ab36a8122d73ab42a3b9f4fc473728454/types/didyoumean/index.d.ts#L3

I think we should add `object[]` to list parameter 
```js
declare function didYouMean(str: string, list: string[] | object[], key?: string): string | string[];
```

**Pull request merged**